.github/workflows/tests.yml

```diff
@@ -0,0 +1,120 @@
+        echo GEF_CACHE_DIR=`python3 -m pip cache dir` >> $GITHUB_ENV
+        echo GEF_PATH_DIR=${HOME}/gef >> $GITHUB_ENV
+        echo GEF_PATH="${HOME}/gef/gef.py" >> $GITHUB_ENV
+
+    - name: Cache dependencies
+      uses: actions/cache@v3
+      id: cache-deps
+      env:
+        cache-name: cache-deps
+      with:
+        key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
+        path: |
+          ${{ env.GEF_CACHE_DIR }}
+        restore-keys:
+          ${{ runner.os }}-pip-${{ env.cache-name }}-
+          ${{ runner.os }}-pip-
+          ${{ runner.os }}-${{ env.cache-name }}-
+          ${{ runner.os }}-
+
+    - name: Install requirements
+      run: |
+        mkdir -p ${{ env.GEF_CACHE_DIR }}
+        python${{ env.PY_VER }} -m pip install --user --upgrade -r ./requirements.txt -r ./tests/requirements.txt
+
+    - name: Checkout GEF
+      run: |
+        mkdir -p ${{ env.GEF_PATH_DIR }}
+        wget -O ${{ env.GEF_PATH }} https://raw.githubusercontent.com/hugsy/gef/${{ env.BRANCH }}/gef.py
+        echo "source ${{ env.GEF_PATH }}" > ~/.gdbinit
+        gdb -q -ex 'gef missing' -ex 'gef help' -ex 'gef config' -ex start -ex continue -ex quit /bin/pwd
+
+    - name: Build config file
+      run: |
+        gdb -q \
+          -ex "gef config pcustom.struct_path '$(pwd)/structs'" \
+          -ex "gef config syscall-args.path '$(pwd)/syscall-tables'" \
+          -ex "gef config context.libc_args True" \
+          -ex "gef config context.libc_args_path '$(pwd)/glibc-function-args'" \
+          -ex 'gef save' \
+          -ex quit
+
+    - name: Run Tests
+      run: |
+        make -C tests/binaries -j ${{ env.NB_CPU }}
+        python${{ env.PY_VER }} -m pytest --forked -n ${{ env.NB_CPU }} -v -k "not benchmark" tests/
+
+    - name: Run linter
+      run: |
+        python${{ env.PY_VER }} -m pylint --rcfile=$(pwd)/.pylintrc gef.py tests/*/*.py
+
+  standalone:
+    runs-on: ubuntu-latest
+    name: "Verify GEF-Extras install from gef/scripts"
+    steps:
+    - name: Install python and toolchain
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y wget gdb-multiarch python3-dev python3-pip python3-wheel python3-setuptools git cmake gcc g++ pkg-config libglib2.0-dev gdbserver
+        sudo python3 -m pip install --upgrade pip
+        echo PY_VER=`gdb -q -nx -ex "pi print('.'.join(map(str, sys.version_info[:2])))" -ex quit` >> $GITHUB_ENV
+        echo NB_CPU=`grep -c ^processor /proc/cpuinfo` >> $GITHUB_ENV
+        echo GEF_CI_ARCH=`uname --processor` >> $GITHUB_ENV
+        echo GEF_CACHE_DIR=`python3 -m pip cache dir` >> $GITHUB_ENV
+        echo GEF_PATH_DIR=${HOME}/gef >> $GITHUB_ENV
+        echo GEF_EXTRAS_PATH=${HOME}/gef-extras >> $GITHUB_ENV
+        echo GEF_PATH="${HOME}/gef/gef.py" >> $GITHUB_ENV
+
+    - name: Checkout GEF & GEF-Extras
+      run: |
+        mkdir -p ${{ env.GEF_PATH_DIR }} ${{ env.GEF_EXTRAS_PATH }}
```

Comment:
I personally don't like to assume that curl is installed... 

sudo apt-get update
sudo apt-get install -y curl

Reply:
Damn it, I meant to replace this with `wget`, forgot that one spot. Good catch